Supervisor visual changes

rae-stanton opened the pull request:
This adjusts the tests to be reflective of their now relevant Bootstrap classes versus the CSS from before. I'm not sure we needed the custom CSS since Bootstrap gives it to us for free, but I am happy to either keep the CSS file or we can remove it and the changes won't be affected from what I can see. 

### What github issue is this PR for, if any?
Resolves #4758 

### What changed, and why?
This refactored the previous visual where all of the classes were inline with each other, versus being on corresponding rows that made a little more sense to have separated based on volunteers and transitioned youth. I also changed the spans to divs. This should also keep everything within a flexbox that doesn't go past the "actions" column and beyond. The second flex row should also stay within a 50% margin overall just for visual sake. 

### How will this affect user permissions?
No permissions were changed, so I guess:
- Supervisor permissions: visual changes
- Admin permissions: visual changes

### How is this tested? (please write tests!) 💖💪
Tests were adjusted to reflect use of bootstrap 5 classes versus the custom CSS from before.

### Screenshots please :)
<img width="956" alt="Screenshot 2023-04-29 at 1 52 56 PM" src="https://user-images.githubusercontent.com/101673900/235324012-dd49e8e9-5d71-4345-8b31-5aafcf2122a3.png">

<img src="https://media0.giphy.com/media/BK1EfIsdkKZMY/giphy.gif"/>

rae-stanton committed:
Supervisor visual changes

This adjusts the tests to be reflective of their now relevant Bootstrap classes versus the CSS from before. I'm not sure we needed the custom CSS since Bootstrap gives it to us for free, but let me know if we want to remove this.

littleforest reviewed (commented):
Thanks for working on this! I'm all for removing the CSS file if it is no longer needed. Can you take a look at the erblint error and see what fixes need to be made?

rae-stanton commented:
@littleforest Absolutely! I think it may be a config issue on my end, I'll update in the AM. 😊

rae-stanton committed:
fixed lint issue

Lint should now be all good! I added commas to mark new lines by accident. Commas are now removed and everything should be lined up nicely.

rae-stanton commented:
@littleforest I totally added extra commas to separate lines when I didn't need to - total oops on my part, but everything should be good to go now! 😊

rae-stanton committed:
WIP test update

Unsure why this part of the test is not passing after most recent commit - currently working on this particular part of the test file to get it green again.

rae-stanton committed:
Updated index for linter

This adjusted the index file that caused linter to fail - all should be ready to go with passing tests and everything.

rae-stanton committed:
Merge branch '#4758-supervisor-visual' of github.com:rae-stanton/casa into #4758-supervisor-visual

rae-stanton commented:
@littleforest Ok this should now all be ready to go! Let me know if you see any additional changes that need to be made. 😊

littleforest commented on spec/views/supervisors/index.html.erb_spec.rb:
```diff
@@ -35,29 +35,29 @@
+      #   expect(inactive_flex).to eq(inactive_content)
+      #   expect(active_flex.to_i).to eq(2)
+      #   expect(inactive_flex.to_i).to eq(1)
+      # end
```
@rae-stanton is this test broken now? Can it be modified to match the new layout?

rae-stanton replied:
@littleforest oops! I thought I committed the final test - it's passing now and I've added the updated commit for the test file.

rae-stanton committed:
test fix

had to make sure the test passed with adjustments; all passing now

rae-stanton committed:
Merge branch 'main' into #4758-supervisor-visual

rae-stanton committed:
Update index.html.erb_spec.rb

rae-stanton committed:
Merge branch '#4758-supervisor-visual' of github.com:rae-stanton/casa into #4758-supervisor-visual

littleforest reviewed (commented):
Could you just run `standardrb --fix` to fix the linting? The spec failure is a known issue and not a problem.

rae-stanton committed:
standard RB

standard RB adjustments

littleforest reviewed (approved):
Thanks for adding tests!
